src/Shared/FileUtilities.cs

```diff
@@ -53,8 +53,8 @@ internal static void ClearCacheDirectoryPath()
             cacheDirectory = null;
         }
 
-        // TODO: assumption on file system case sensitivity: https://github.com/Microsoft/msbuild/issues/781
-        internal static readonly StringComparison PathComparison = StringComparison.OrdinalIgnoreCase;
+        internal static readonly StringComparison PathComparison = Directory.Exists(Path.GetTempPath().ToUpper()) && Directory.Exists(Path.GetTempPath().ToLower()) ?
```

Comment:
Might want to cache this, too.

@rainersigwald

Reply:
It's a static field, it will be executed once per type load right?

Reply:
Good point. I was really thinking about if we wanted to do something more complicated/robust like:
```
internal static StringComparison PathComparison2 { get {
    if (_pathComparison == null)
    {
        string pathName = Path.GetTempFileName();
        _pathComparison = File.GetLastWriteTime(pathName) == File.GetLastWriteTime(pathName.ToUpper()) ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
    }
    return _pathComparison.Value;
} }
        private static StringComparison? _pathComparison;
```

Reply:
At that point putting it in a Lazy would probably be easier. But let's start simple and complicate it if/when necessary.